ksqldb-serde/src/main/java/io/confluent/ksql/serde/connect/ConnectKsqlSchemaTranslator.java

```diff
@@ -184,6 +184,7 @@ private Schema toKsqlStructSchema(final Schema schema) {
         }
       } catch (final UnsupportedTypeException e) {
         log.error("Error inferring schema at field {}: {}", field.name(), e.getMessage());
+        throw e;
```

Comment:
Is this change related to KLIP-56? What's the motivation for this change?

Also, this change is backwards incompatible, right? I think the impact of this change is that fields with certain types of incompatible schemas were ignored while now the entire statement would be rejected. Is that correct? AFAICT this method is only ever called from the DefaultSchemaInjector which should only be called when new statements are submitted, and not when existing statements are executed from the command topic. Can you confirm? (This is important so we don't break compatibility for existing statements that have already been written to the command topic.)

Reply:
Should be related to KLIP-56 as this could be thrown during schema inference.

Yeah. I can only find this is called in `DefaultSchemaInjector` before command is written to command topic. 

> I think the impact of this change is that fields with certain types of incompatible schemas were ignored while now the entire statement would be rejected. Is that correct?

Yes. I want to make the unsupported type error explicit instead of silently ignored and make user think their schemas are inferred correctly but in fact not